### DISCUSSION TITLE
failed to add device /sys/devices/pci0000:00/0000:00:03.2/0000:03:00.0: EEPROM size invalid, got 0x80000, expected 0x40000

### DISCUSSION THREAD

**Discussion by xiaochunlee** [fwupd/fwupd#5880] [General]
Hello, Richard
We are running fwupd on RHEL9.0/RHEL9.1 & SLES15SP5, but the below error comes.

Error message pop out like:
"fwupd[6351]: 02:13:05.350 FuEngine             failed to add device /sys/devices/pci0000:00/0000:00:03.2/0000:03:00.0: EEPROM size invalid, got 0x80000, expected 0x40000"

fwupd version: 1.8.15
kernel version: 5.14-70.22.1.el9_0.x86_64

The error only comes while scan device 0000:03:00.0, the other devices on the same bridge do not show this errors.
-->>
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens5f0 -> ../../devices/pci0000:00/0000:00:03.2/0000:03:00.0/net/ens5f0
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens5f1 -> ../../devices/pci0000:00/0000:00:03.2/0000:03:00.1/net/ens5f1
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens5f2 -> ../../devices/pci0000:00/0000:00:03.2/0000:03:00.2/net/ens5f2
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens5f3 -> ../../devices/pci0000:00/0000:00:03.2/0000:03:00.3/net/ens5f3
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens9f0np0 -> ../../devices/pci0000:40/0000:40:01.1/0000:41:00.0/net/ens9f0np0
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 ens9f1np1 -> ../../devices/pci0000:40/0000:40:01.1/0000:41:00.1/net/ens9f1np1
lrwxrwxrwx. 1 root root 0 Jun  6 14:21 lo -> ../../devices/virtual/net/lo
<<--

Attach the lspci information for your reference.
-->>
03:00.0 Ethernet controller [0200]: Broadcom Inc. and subsidiaries NetXtreme BCM5719 Gigabit Ethernet PCIe [14e4:1657] (rev 01)
	Subsystem: Lenovo Device [17aa:402d]
	Physical Slot: 5
	Control: I/O- Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
	Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
	Latency: 0, Cache Line Size: 64 bytes
	Interrupt: pin A routed to IRQ 72
	NUMA node: 0
	IOMMU group: 26
	Region 0: Memory at 210020fb0000 (64-bit, prefetchable) [size=64K]
	Region 2: Memory at 210020fa0000 (64-bit, prefetchable) [size=64K]
	Region 4: Memory at 210020f90000 (64-bit, prefetchable) [size=64K]
	Expansion ROM at f18c0000 [disabled] [size=256K]
	Capabilities: [48] Power Management version 3
		Flags: PMEClk- DSI- D1- D2- AuxCurrent=0mA PME(D0+,D1-,D2-,D3hot+,D3cold+)
		Status: D0 NoSoftRst+ PME-Enable- DSel=0 DScale=1 PME-
	Capabilities: [50] Vital Product Data
		Product Name: Broadcom NetXtreme PCIe 1Gb 4-Port RJ45 Ethernet Adapter
		Read-only fields:
			[FN] FRU: 00YK551
			[PN] Part number: SN30L21972
			[SN] Serial number: L2NV94800YM
			[RV] Reserved: checksum good, 88 byte(s) reserved
		Read/write fields:
			[V0] Vendor specific: 222.0.2.1
			[RW] Read-write area: 173 byte(s) free
		End
	Capabilities: [58] MSI: Enable- Count=1/8 Maskable- 64bit+
		Address: 0000000000000000  Data: 0000
	Capabilities: [a0] MSI-X: Enable+ Count=17 Masked-
		Vector table: BAR=4 offset=00000000
		PBA: BAR=4 offset=00001000
	Capabilities: [ac] Express (v2) Endpoint, MSI 00
		DevCap:	MaxPayload 256 bytes, PhantFunc 0, Latency L0s <4us, L1 <64us
			ExtTag- AttnBtn- AttnInd- PwrInd- RBE+ FLReset+ SlotPowerLimit 75.000W
		DevCtl:	CorrErr+ NonFatalErr+ FatalErr+ UnsupReq-
			RlxdOrd- ExtTag- PhantFunc- AuxPwr+ NoSnoop- FLReset-
			MaxPayload 256 bytes, MaxReadReq 512 bytes
		DevSta:	CorrErr- NonFatalErr- FatalErr- UnsupReq- AuxPwr+ TransPend-
		LnkCap:	Port #0, Speed 5GT/s, Width x4, ASPM L0s L1, Exit Latency L0s <1us, L1 <2us
			ClockPM+ Surprise- LLActRep- BwNot- ASPMOptComp-
		LnkCtl:	ASPM Disabled; RCB 64 bytes, Disabled- CommClk+
			ExtSynch- ClockPM- AutWidDis- BWInt- AutBWInt-
		LnkSta:	Speed 5GT/s (ok), Width x4 (ok)
			TrErr- Train- SlotClk+ DLActive- BWMgmt- ABWMgmt-
		DevCap2: Completion Timeout: Range ABCD, TimeoutDis+ NROPrPrP- LTR-
			 10BitTagComp- 10BitTagReq- OBFF Not Supported, ExtFmt- EETLPPrefix-
			 EmergencyPowerReduction Not Supported, EmergencyPowerReductionInit-
			 FRS- TPHComp- ExtTPHComp-
			 AtomicOpsCap: 32bit- 64bit- 128bitCAS-
		DevCtl2: Completion Timeout: 50us to 50ms, TimeoutDis- LTR- OBFF Disabled,
			 AtomicOpsCtl: ReqEn-
		LnkCtl2: Target Link Speed: 2.5GT/s, EnterCompliance- SpeedDis-
			 Transmit Margin: Normal Operating Range, EnterModifiedCompliance- ComplianceSOS-
			 Compliance De-emphasis: -6dB
		LnkSta2: Current De-emphasis Level: -6dB, EqualizationComplete- EqualizationPhase1-
			 EqualizationPhase2- EqualizationPhase3- LinkEqualizationRequest-
			 Retimer- 2Retimers- CrosslinkRes: unsupported
	Capabilities: [100 v1] Advanced Error Reporting
		UESta:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UEMsk:	DLP- SDES- TLP- FCP- CmpltTO- CmpltAbrt- UnxCmplt- RxOF- MalfTLP- ECRC- UnsupReq- ACSViol-
		UESvrt:	DLP+ SDES+ TLP- FCP+ CmpltTO- CmpltAbrt- UnxCmplt- RxOF+ MalfTLP+ ECRC- UnsupReq- ACSViol-
		CESta:	RxErr- BadTLP- BadDLLP- Rollover- Timeout- AdvNonFatalErr+
		CEMsk:	RxErr- BadTLP- BadDLLP- Rollover- Timeout- AdvNonFatalErr+
		AERCap:	First Error Pointer: 00, ECRCGenCap+ ECRCGenEn+ ECRCChkCap+ ECRCChkEn-
			MultHdrRecCap- MultHdrRecEn- TLPPfxPres- HdrLogCap-
		HeaderLog: 00000000 00000000 00000000 00000000
	Capabilities: [13c v1] Device Serial Number 00-00-b0-26-28-92-2d-18
	Capabilities: [150 v1] Power Budgeting <?>
	Capabilities: [160 v1] Virtual Channel
		Caps:	LPEVC=0 RefClk=100ns PATEntryBits=1
		Arb:	Fixed- WRR32- WRR64- WRR128-
		Ctrl:	ArbSelect=Fixed
		Status:	InProgress-
		VC0:	Caps:	PATOffset=00 MaxTimeSlots=1 RejSnoopTrans-
			Arb:	Fixed- WRR32- WRR64- WRR128- TWRR128- WRR256-
			Ctrl:	Enable+ ID=0 ArbSelect=Fixed TC/VC=01
			Status:	NegoPending- InProgress-
	Kernel driver in use: tg3
	Kernel modules: tg3
<<--

**Comment by xiaochunlee** [comment]
Command "ethtool -e ens5f0" and "ethtool -e ens5f1"  show the eeprom length is 0x80000 on  four net ports of BCM5719. But our code seems thinks the maximum length is 0x40000

**Comment by hughsie** [comment]
https://github.com/fwupd/fwupd/pull/5882 should be the fix we want to include, the first commit being the one we should backport.

**Reply by xiaochunlee** [reply]
Really appreciated it, will you build a RHEL rpm package for us test?

**Reply by hughsie** [reply]
Can you test https://people.redhat.com/rhughes/packages/fwupd/el9/ please? If that doesn't work don't panic.

**Reply by xiaochunlee** [reply]
Yes, it works well, there isn't any errors of fwupd on our systems! Thanks!

**Reply by hughsie** [reply]
Excellent. One request. Can you get me the output of `sudo fwupdtool --plugins bcm57xx --verbose --verbose get-devices` please -- and also add your testing notes to the RH bug. Thanks.

**Reply by xiaochunlee** [reply]
Yes, I had updated your request to the RH bug!

**Reply by hughsie** [reply]
Thanks; lets move planning there.
